source/renderer/app/components/layout/TopBar.js

```diff
@@ -26,13 +27,18 @@ export default class TopBar extends Component<Props> {
     ]);
 
     const topBarTitle = activeWallet ? (
-      <div className={styles.walletInfo}>
-        <div className={styles.walletName}>{activeWallet.name}</div>
-        <div className={styles.walletAmount}>
+      <span className={styles.walletInfo}>
+        <span className={styles.walletName}>{activeWallet.name}</span>
+        <span className={styles.walletAmount}>
           {// show currency and use long format
           formattedWalletAmount(activeWallet.amount, true)}
-        </div>
-      </div>
+        </span>
+        {activeWallet.isLegacy && (
+          <div className={styles.legacyBadge}>
```

Comment:
@yakovkaravelov why do we need a `<div className={styles.legacyBadge}>` wrapper here? Can't we move all the needed HTML elements into the `<LegacyBadge />` itself?

Reply:
@nikolaglumac 
This is because we need to position legacy badge in different ways in places.
For example positioning it in sidebar menu and topbar title sections are different and that's why they use wrapper element to style accordingly.

Reply:
@yakovkaravelov it's better to keep the logic inside the component. It could receive some props for positioning. Maybe only left and right would be enough. Otherwise, an object with the style needed.